pkg/controller/direct/alloydb/cluster_controller.go

```diff
@@ -28,30 +28,23 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
 
 	krm "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/alloydb/v1beta1"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
 )
 
-// AddClusterController creates a new controller and adds it to the Manager.
-// The Manager will set fields on the Controller and start it when the Manager is started.
-func AddClusterController(mgr manager.Manager, config *controller.Config, opts directbase.Deps) error {
-	gvk := krm.AlloyDBClusterGVK
+func init() {
+	directbase.ControllerBuilder.RegisterModel(krm.AlloyDBClusterGVK, NewModel)
+}
 
-	// TODO: Share gcp client (any value in doing so)?
-	ctx := context.TODO()
-	gcpClient, err := newGCPClient(ctx, config)
-	if err != nil {
-		return err
-	}
-	m := &clusterModel{gcpClient: gcpClient}
-	return directbase.Add(mgr, gvk, m, opts)
+func NewModel(config *controller.Config) directbase.Model {
```

Comment:
Nit: We'll probably end up naming this NewClusterModel (or newClusterModel).  But not a blocker.

Reply:
I like the private method name.